### ISSUE TITLE
Remove the opentelemetry-api dependency from the opentelemetry-semconv JAR

### ISSUE THREAD

**Issue by marcingrzejszczak** [open-telemetry/opentelemetry-java#4477]
**Is your feature request related to a problem? Please describe.**
If one wants to just use the semantic conventions JAR, out of the box one gets `opentelemetry-context` and `opentelemetry-api` JARs which seem to be completely redundant. The only thing that is being used is the `AttributeKey` together with its implementation.

**Describe the solution you'd like**
I would prefer the `AttributeKey` and the implementation to be copied to `opentelemetry-semconv` so that it doesn't have any additional dependencies. The idea behind OTLP was to generate OTel compliant output without necessarily using the OTel libraries. We would like to achieve the similar outcome - having OTel compliant tags by using the OTel semantic conventions but not necessarily by using other OTel libraries.

**Additional context**
There might be people that want to follow the OpenTelemetry semantic conventions of tags, but not necessarily use the OpenTelemetry tracer. We're considering using this in Micrometer but we don't want to take in the other JARs.

**Comment by anuraaga** [comment]
Hi @marcingrzejszczak - for a few reasons it would be logistically difficult for us to move `AttributeKey` to this artifact or remove that coupling.

The artifact is just generated from a yaml file would you be able to generate into your own code with customizations that you need?

https://github.com/open-telemetry/opentelemetry-java/blob/main/buildscripts/semantic-convention/generate.sh

**Comment by marcingrzejszczak** [comment]
Yeah - that's what we were assuming (that it might be hard) and what we were planning to eventually do (generate it ourselves)

**Comment by jack-berg** [comment]
Closing as `wontfix` for the reasons stated above.
